src/Sarif.FunctionalTests/v2/ConverterTestData/AndroidStudio/AndroidStudio_issueLog5_raw.xml.sarif

```diff
@@ -1,50 +1,12 @@
 {
-  "$schema": "http://json.schemastore.org/sarif-2.0.0-csd.2.beta.2018-09-26",
-  "version": "2.0.0-csd.2.beta.2018-09-26",
+  "$schema": "http://json.schemastore.org/sarif-2.0.0-csd.2.beta.2018-11-28",
+  "version": "2.0.0-csd.2.beta.2018-11-28",
```

Comment:
>version [](start = 3, length = 7)

OK, mysterious. It turns out that this file, of all the converter test files, is the _only one_ that contains multiple results that are identical. This report file apparently finds multiple occurrences of typos on the same line. Only the line # is reported. Previously, our code only emitted a single result for this case. Now we're getting two. I have verified the source raw android studio has this duplication. I can't yet account for how/why this duplicated result wasn't getting added to the log file. Looking.

Reply:
OK, figured it out. We made a bad change in the android studio converter, which was previously eliminating issue duplicated via a hashset. We now pass a list directly to some loggers, so this conversion to list allowed multiple identical issues to be generated. The results array, in fact, s/be unique. What could or should have happened here is that we added a second, duplicated location to the common result. Showing, we have one result, two occurrences which have identical locations.

I will open an SDK issue about preferring hashsets over lists for persisting results. 

---
In reply to: [248454813](https://github.com/Microsoft/sarif-sdk/pull/1217#discussion_r248454813) [](ancestors = 248454813)